### PR TITLE
Create cart app to manage shopping cart

### PR DESCRIPTION
**Pull request by Karthikeyantestpress**

### CODE REVIEW

File: myshop/tests/cart/test_cart.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+from tests.cart.test_model_mixin_testcase import ModelMixinTestcase
+
+
+class Testurls(ModelMixinTestcase, TestCase):
+    def test_initialize_cart_clean_session(self):
+
+        self.assertEqual(self.cart_obj.cart, {})
+
+    def test_add_new_product_to_the_cart(self):
+        self.cart_obj.add_product(self.newproduct, quantity=3)
```
Comment (waranlogesh) [inline myshop/tests/cart/test_cart.py]:
```suggestion
        self.cart.add_product(self.newproduct, quantity=3)

```
Don't add too much unnecessary detail in the name. It affects the readability

File: myshop/tests/cart/test_cart.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        cart_list = self.request.session.get(settings.CART_SESSION_ID)
+
+        self.assertEqual(cart_list["1"]["quantity"], 3)
```
Comment (waranlogesh) [inline myshop/tests/cart/test_cart.py]: Assert from self.cart itself. No need to take it from session and store it in new variable and add complexity

Comment (waranlogesh) [reply myshop/tests/cart/test_cart.py]: Same for all the below test cases

File: myshop/tests/cart/test_model_mixin_testcase.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        self.request.session.save()
+
+        self.cart_obj = Cart(self.request)
+
+        self.newcategory = Category.objects.create(
+            name="Gaming",
+            slug="gaming",
+        )
+
+        self.newproduct = Product.objects.create(
```
Comment (waranlogesh) [inline myshop/tests/cart/test_model_mixin_testcase.py]:
```suggestion
        self.product = Product.objects.create(

```

File: myshop/tests/cart/test_model_mixin_testcase.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+class ModelMixinTestcase(TestCase):
+    def setUp(self):
+        self.request = RequestFactory().get("/")
+        middleware = SessionMiddleware(get_response=self.request)
+        middleware.process_request(self.request)
+        self.request.session.save()
+
+        self.cart_obj = Cart(self.request)
+
+        self.newcategory = Category.objects.create(
```
Comment (waranlogesh) [inline myshop/tests/cart/test_model_mixin_testcase.py]:
```suggestion
        self.category = Category.objects.create(

```

File: myshop/tests/cart/test_model_mixin_testcase.py [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+
+class ModelMixinTestcase(TestCase):
+    def setUp(self):
+        self.request = RequestFactory().get("/")
+        middleware = SessionMiddleware(get_response=self.request)
+        middleware.process_request(self.request)
+        self.request.session.save()
+
+        self.cart_obj = Cart(self.request)
```
Comment (waranlogesh) [inline myshop/tests/cart/test_model_mixin_testcase.py]:
```suggestion
        self.cart = Cart(self.request)

```

**Review by waranlogesh — CHANGES_REQUESTED** [review]

File: myshop/tests/cart/test_cart.py (hunk 2 above)
Comment (Karthikeyantestpress) [reply myshop/tests/cart/test_cart.py]: Separated the PR and updated the changes suggested in other new PR
https://github.com/Karthikeyantestpress/Online_Shopping/pull/15

File: myshop/tests/cart/test_model_mixin_testcase.py (hunk 3 above)
Comment (Karthikeyantestpress) [reply myshop/tests/cart/test_model_mixin_testcase.py]: Separated the PR and updated the changes suggested in other new PR
https://github.com/Karthikeyantestpress/Online_Shopping/pull/15

File: myshop/tests/cart/test_model_mixin_testcase.py (hunk 4 above)
Comment (Karthikeyantestpress) [reply myshop/tests/cart/test_model_mixin_testcase.py]: Separated the PR and updated the changes suggested in other new PR
https://github.com/Karthikeyantestpress/Online_Shopping/pull/15

File: myshop/tests/cart/test_model_mixin_testcase.py (hunk 5 above)
Comment (Karthikeyantestpress) [reply myshop/tests/cart/test_model_mixin_testcase.py]: Separated the PR and updated the changes suggested in other new PR
https://github.com/Karthikeyantestpress/Online_Shopping/pull/15

**Pull request merged**